k9mail/src/main/java/com/fsck/k9/activity/setup/AccountSettings.java

```diff
@@ -728,9 +729,22 @@ private void setupCryptoStuff() {
         autocryptPreferEncryptMutual = findPreference(PREFERENCE_AUTOCRYPT_PREFER_ENCRYPT);
         pgpHideSignOnly = (SwitchPreference) findPreference(PREFERENCE_CRYPTO_HIDE_SIGN_ONLY);
 
-        boolean isPgpConfigured = account.isOpenPgpProviderConfigured();
+        String pgpProvider = account.getOpenPgpProvider();
+        String pgpProviderName = null;
+        boolean isPgpConfigured = pgpProvider != null;
```

Comment:
This duplicates the code in `Account.isOpenPgpProviderConfigured()`. Why not keep the original call?

Reply:
Because I need the pgpProvider value, and AS will complain that pgpProvider might be null if I don't do the check here explicitly

Reply:
changed after all, to use `isOpenPgpProviderConfigured()` everywhere